pkg/agent/containerd/containerd.go

```diff
@@ -122,5 +127,41 @@ func Run(ctx context.Context, cfg *config.Node) error {
 		}
 	}
 
+	imageDir := "/var/lib/rancher/k3s/agent/images"
+	fileInfo, err := os.Stat(imageDir)
+	if err != nil {
+		logrus.Infof("Cannot find images in %s: %v", imageDir, err)
+	} else {
+		if fileInfo.IsDir() {
+			fileInfos, err := ioutil.ReadDir(imageDir)
+			if err != nil {
+				logrus.Infof("Cannot read images in %s: %v", imageDir, err)
+			}
+			client, err := containerd.New(cfg.Containerd.Address)
+			if err != nil {
+				return err
+			}
+			defer client.Close()
+
+			ctxContainerD := namespaces.WithNamespace(context.Background(), "k8s.io")
+
+			for _, fileInfo := range fileInfos {
+				if !fileInfo.IsDir() {
+					filePath := filepath.Join(imageDir, fileInfo.Name())
+					fileContent, err := ioutil.ReadFile(filePath)
```

Comment:
This is going to cause memory issues because you are reading the whole thing into memory (could be gigabytes).  Can you just open the file and pass that to import?